src/index.ts

```diff
@@ -1169,43 +1169,49 @@ export class KycDao extends ApiBase {
           (network) => Number(networkDetails[network].chainId) === Number(chainId),
         );
 
-        // selected network is not supported
-        if (!blockchainNetwork) {
-          throw new StatusError('NetworkNotSupported');
-        }
-
         // selected network is not enabled
-        if (!this.blockchainNetworks.includes(blockchainNetwork)) {
+        if (!blockchainNetwork || !this.blockchainNetworks.includes(blockchainNetwork)) {
           // automatic network switching is enabled and there is only one enabled network
           if (autoSwitch && this.blockchainNetworks.length === 1) {
             const enabledNetwork = this.blockchainNetworks[0];
             const enabledNetworkDetails = networkDetails[enabledNetwork];
 
-            // the enabled network has the same chain as the selected one and it has a chain ID
-            if (
-              enabledNetworkDetails.blockchain === networkDetails[blockchainNetwork].blockchain &&
-              enabledNetworkDetails.chainId
-            ) {
+            // the enabled network has 'Ethereum' chain and it has a chain ID
+            if (enabledNetworkDetails.blockchain === 'Ethereum' && enabledNetworkDetails.chainId) {
               const newChainId = enabledNetworkDetails.chainId;
               try {
                 await this.evmProvider.switchNetwork(newChainId);
 
                 const updatedChainId = await this.evmProvider.getChainId();
                 if (updatedChainId !== newChainId) {
-                  throw new InternalError('ChainId expected to be changed.');
+                  throw new StatusError(
+                    'NetworkSwitchingFailed',
+                    'Automatic EVM network switching failed: chain ID did not change.',
+                  );
```

Comment:
This was an internal error before, because if the network switching would fail, then the switchNetwork call a few lines above should throw an exception. So this is just kind of a sanity check here, if the switching to a chainId is marked as successful by the wallet, then the chainId after that call would be the one requested by switchWallet.

Reply:
Ok, I'll change it back since logically I agree kinda, but my other logic here was that we are explicitly handling this error here and from the user's standpoint the network switching clearly failed, they have to do the same thing, switch the wallet manually. The UI can handle this issue based on error type, while if we throw an OMG InternalError, then the flow will fail and a user will probably get stuck on this issue.